### DISCUSSION TITLE
Can I add data preprocessor to the exported model ?

### DISCUSSION THREAD

**Discussion by CescMessi** [open-mmlab/mmdeploy#1644] [Q&A]
In mmengine, data preprocessor and model forward are separated, resulting in models exported by mmdeploy that do not contain data preprocessor. I had to manually add the normalization part when using the exported model. I know that rewriting the forward function can do it, but it will result in repeated normalization when exporting. Is there any easier way to do that?

**Comment by lvhan028** [comment]
mmdeploy SDK already supports that. 
You can use `--dump-info` when exporting a model, and use mmdeploy SDK's API to do model inference. 
Introduction can be found from [here](https://mmdeploy.readthedocs.io/en/latest/get_started.html#)
